cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -419,8 +466,8 @@ private LocalFile uploadCacheableFile(String bucket, String targetFilePath,
   /**
    * Uploads files to gcs.
    */
-  private LocalFile uploadFile(String bucket, String targetFilePath,
-                               LocalFile localFile, boolean isCacheable)
+  private LocalFile uploadFile(String bucket, String targetFilePath, LocalFile localFile,
```

Comment:
why do we need isCacheablePerVersion? isCacheable parameter seems to be enough.

Reply:
On line no 429 - https://github.com/cdapio/cdap/pull/14650/files#diff-b7675d51d200ad4368d96b266bc04c96d4ecd20e5d1e1a6510f43fee970ffacbL429

We set the custom time on the blobs if they are cachable which is not required for these set of artifacts, so we need an extra flag to separate these uploads.

Reply:
Had a second thought, was able to workaround it using the list of artifacts.